### ISSUE TITLE
RFC: include debug version for tracers in the git repository

### ISSUE THREAD

**Issue by florianl** [open-telemetry/opentelemetry-ebpf-profiler#225]
This RFC originates from a [discussion on slack](https://cloud-native.slack.com/archives/C03J794L0BV/p1730392737850669).

[Currently](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/commit/0c58efd41280bce80cca9804896c76874d383b35) only release ELFs for the eBPF programs checked into the repository. `support/ebpf/tracer.ebpf.release.amd64` accounts for 291 KBytes and `support/ebpf/tracer.ebpf.release.arm64` for 283 KBytes. Every time something changes in the eBPF programs, these ELFs need to be updated and will account for some space in the git repository.
When using this repository as dependency, it is [currently](https://github.com/open-telemetry/opentelemetry-ebpf-profiler/commit/0c58efd41280bce80cca9804896c76874d383b35) not possible without additional steps to load the eBPF programs with debug symbols, as they do not exist by default.

If the eBPF programs with debug information will be added to the repository, the size of the repository will increase. At the moment, `support/ebpf/tracer.ebpf.debug.amd64` does have a size of 1.4 MBytes and `support/ebpf/tracer.ebpf.debug.arm64` also does have a size of 1.4 MBytes.

Some previous and related discussions were also happening in https://github.com/open-telemetry/opentelemetry-ebpf-profiler/pull/145. Other OTel SIGs do have similar discussions ongoing - e.g. https://github.com/open-telemetry/opentelemetry-go-instrumentation/issues/1233

For visibility - friendly ping to people that were involved in the discussion on slack: @umanwizard @Gandem @rockdaboot @athre0z @dmathieu

**Comment by christos68k** [comment]
Based on previous discussions, I think the consensus is not to store debug tracers in the repository. We might be able to get away with storing release tracers as their size is comparatively small but we should monitor performance and adapt to a different solution if that's problematic.

**Comment by tsint** [comment]
Since tracers cannot yet be integrated as a package by third-party software, I support not storing debug tracers to the repository. However, I do suggest that build with debug tracer by default, without requiring an additional `make debug` to build.

**Comment by rockdaboot** [comment]
@tsint Can you open a separate issue for the " I do suggest that build with debug tracer by default, without requiring an additional make debug to build."?

**Comment by tsint** [comment]
> @tsint Can you open a separate issue for the " I do suggest that build with debug tracer by default, without requiring an additional make debug to build."?

OK, I'll prepare it and create a new RFC at the right time.

**Comment by christos68k** [comment]
Can we close this issue?

**Comment by efd6** [comment]
Not including all required compilation source (this includes any files referenced by a `//go:embed` compiler directive) for _all_ possible build constraints breaks code vendoring. This has implications for reproducible builds (they are not guaranteed since the blob is free to move depending on the build) and significantly it removes the option for client code that as to be used in a high security posture setting where all artefacts must be vetted.

**Comment by efd6** [comment]
> In theory, there could be thousands of different builds flags with the same number of blobs,

No, this is not the case without appeal to hyperbole; the Go source includes embed directives that state to the build system that the blobs exist on the local file system. This is a contract that is committed to. That contract is made use of by vendoring, which must make available all the source (including embeded files which are source for the purposes here) at the time of vendoring so that builds that use them can be performed.

> and it doesn't seem practical to include them all into a single application binary.

They don't all end up in the application binary. If they did, this issue would not arise since the blobs would already be committed to the repo since their absence would break go get builds without vendoring.
